order/views.py

```diff
@@ -1,3 +1,137 @@
-from django.shortcuts import render
+import json
+from random import randint
 
-# Create your views here.
+from django.views import View
+from django.http import JsonResponse
+
+from .models import Order, OrderList, OrderStatus
+from product.models import Product, Color, ProductColor, Size, ProductSize
+from user.models import User
+from core.utils import login_decorator
+
+def delivery_fee_calculator(orderlists):
+    sum = 0
+    for orderlist in orderlists:
+        sum += orderlist.product.price*orderlist.quantity
+    if sum > 50000:
+        orderlist.order.delivery_fee = 0
+    else:
+        orderlist.order.delivery_fee = 2500
+    orderlist.order.save()
+
+class OrderListView(View):
+    @login_decorator
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            user     = request.user
+            product  = Product.objects.get(pk=data["product_id"])
+            quantity = data["quantity"]
+            color_id = data.get("color_id",None)
+            size_id  = data.get("size_id",None)
+            order    = Order.objects.filter(user=user, orderstatus_id=1)
+
+            if quantity < 1:
+                return JsonResponse({"message":"INVALID_QUANTITY"},status=400)
+            if not order.exists():
+                created_order = Order.objects.create(
+                    user           = user,
+                    orderstatus_id = 1
+                )
+                year          = str(created_order.updated_at.year)
+                month         = str(created_order.updated_at.month)
+                day           = str(created_order.updated_at.day)
+                random_number = str(randint(10000000,99999999))
+                number        = year+month+day+random_number
+                created_order.number = number
+                created_order.save()
+
+            current_order = order.last()
+            orderlist, created = OrderList.objects.get_or_create(
+                product=product,
+                order=current_order,
+                color_id=color_id,
+                size_id=size_id
+            )
+            if created:
+                orderlist.quantity = 0
+            orderlist.quantity += quantity
+            orderlist.save()
+
+            orderlists = current_order.orderlist_set.all()
+            delivery_fee_calculator(orderlists)
+            return JsonResponse({"message":"SUCCESS"}, status=201)
+        except Product.DoesNotExist:
+            return JsonResponse({"message":"INVALID_PRODUCT"}, status=400)
+        except User.DoesNotExist:
+            return JsonResponse({"message":"INVALID_USER"}, status=400)
+        except KeyError:
+            return JsonResponse({"message":"INVALID_KEYS"}, status=400)
+    
+    @login_decorator
+    def get(self, request):
+        orderlists = Order.objects.filter(user=request.user, orderstatus_id=1).last().orderlist_set.select_related('product__seller', 'product__delivery','size','color').all()
```

Comment:
```suggestion
        orderlists = Order.objects.get(user=request.user, orderstatus_id=1).orderlist_set.select_related('product__seller', 'product__delivery','size','color').all()
```
어차피 대상이 하나뿐이므로 get을 사용해주셔도 좋습니다 :)

Reply:
대신 Doesnot exist 에러는 잡아주셔야 할 것 같습니다

Reply:
수정했습니다!